cpmpy/expressions/globalconstraints.py

```diff
@@ -285,3 +287,38 @@ def value(self):
     def __repr__(self):
         return "{}[{}]".format(self.args[0], self.args[1])
 
+class XOR(GlobalConstraint):
+    """
+        The 'xor' constraint for more then 2 arguments.
+        Acts like cascaded xor operators with two inputs
+    """
+
+    def __init__(self, arg_list):
+        if len(arg_list) == 2 and is_num(arg_list[1]):
+            arg_list[0], arg_list[1] = arg_list[1], arg_list[0]
+        i = 0  # length can change
+        while i < len(arg_list):
+            if isinstance(arg_list[i], GlobalConstraint) and arg_list[i].name == "xor":
+                # merge args in at this position
+                arg_list[i:i + 1] = arg_list[i].args
+            else:
+                i += 1
+        super().__init__("xor", arg_list)
+
+    def decompose(self):
+        if len(self.args) == 2:
+            return (self.args[0] + self.args[1]) == 1
+        prev_var, cons = get_or_make_var(self.args[0] ^ self.args[1])
+        for arg in self.args[2:]:
+            prev_var, new_cons = get_or_make_var(prev_var ^ arg)
+            cons += new_cons
+        return cons + [prev_var]
+
+    def value(self):
+        return sum(argval(a) for a in self.args) % 2 == 1
+
+    def __repr__(self):
+        if len(self.args) == 2:
+            return "{} xor {}".format(*self.args)
+        return "xor({})".format(self.args)
+
```

Comment:
does it require a change for the deepcopy? this is in master now

Reply:
Yes, does not follow Globalconstraint constructor syntax, so needs its own `_deepcopy` method